Chapter5/dbServer.erl

```diff
@@ -0,0 +1,72 @@
+-module(dbServer).
+-export([start/0, stop/0, write/2, delete/1, read/1, match/1]).
+-export([loop/1]).
+
+%%Interface
+start() -> register(db, spawn(dbServer, loop, [[]])).
+
+write	(Key, Element) 	-> call(write, 	{Key, Element}	).
+delete	(Key) 			-> call(delete, Key				).
+read	(Key) 			-> call(read, 	Key				).
+match	(Element) 		-> call(match, 	Element			).
+
+stop() ->
+	db ! {stop, self()},
+	receive {reply, Reply} -> Reply end.
+
+call(Type, Message) ->
+	db ! {Type, self(), Message},
+	receive {reply, Reply} -> Reply end.
+
+%%Backend
+loop(Data) ->
+	receive
+		{stop, From} ->
+			reply(From, ok);
+		{Type, From, Msg} ->
+			{Reply, NewData} = handle_msg(Type, Msg, Data),
+			reply(From, Reply),
+			loop(NewData)
+	end.
+
+reply(To, Msg) ->
+	To ! {reply, Msg}.
+
+
+
+
+
+
+
+handle_msg(Type, Message, Data) ->
+	case Type of
+		write ->
+			{Key, Element} = Message,
+			{ok, Data ++ [{Key, Element}]};
```

Comment:
See performance caveat that I [mentioned](https://github.com/danieljharris/learning-erlang/pull/1#discussion_r132897832) earlier.